cmd/aws/create_test.go

```diff
@@ -0,0 +1,402 @@
+					},
+				},
+			},
+			paginateErr: nil,
+			expected:    []string{"t2.micro", "t2.small"},
+			expectedErr: nil,
+		},
+		{
+			name:         "pagination error",
+			architecture: "x86_64",
+			paginateErr:  errors.New("pagination failed"),
+			expected:     nil,
+			expectedErr:  fmt.Errorf("failed to load next pages for instance types: pagination failed"),
+		},
+		{
+			name:         "no matching instance types",
+			architecture: "arm64",
+			instanceTypes: []ec2Types.InstanceTypeInfo{
+				{
+					InstanceType: ec2Types.InstanceTypeT2Micro,
+					ProcessorInfo: &ec2Types.ProcessorInfo{
+						SupportedArchitectures: []ec2Types.ArchitectureType{ec2Types.ArchitectureTypeX8664},
+					},
+				},
+			},
+			paginateErr: nil,
+			expected:    []string(nil),
+			expectedErr: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			paginator := &mockInstanceTypesPaginator{
+				instanceTypes: tt.instanceTypes,
+				err:           tt.paginateErr,
+			}
+
+			got, err := GetSupportedInstanceTypes(context.Background(), paginator, tt.architecture)
+			if tt.expectedErr != nil {
+				assert.EqualError(t, err, tt.expectedErr.Error())
+				assert.Nil(t, got)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expected, got)
+			}
+		})
+	}
+}
+
+type mockInstanceTypesPaginator struct {
+	instanceTypes []ec2Types.InstanceTypeInfo
+	err           error
+	called        bool
+}
+
+func (m *mockInstanceTypesPaginator) HasMorePages() bool {
+	return !m.called
+}
+
+func (m *mockInstanceTypesPaginator) NextPage(ctx context.Context, opts ...func(*ec2.Options)) (*ec2.DescribeInstanceTypesOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+	m.called = true
+	return &ec2.DescribeInstanceTypesOutput{
+		InstanceTypes: m.instanceTypes,
+	}, nil
+}
+
+type mockSSMClient struct {
+	ssm.Client
+	parameterValue string
+	err            error
+}
+
+func (m *mockSSMClient) GetParameter(ctx context.Context, input *ssm.GetParameterInput, opts ...func(*ssm.Options)) (*ssm.GetParameterOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+	return &ssm.GetParameterOutput{
+		Parameter: &ssmTypes.Parameter{
+			Value: &m.parameterValue,
+		},
+	}, nil
+}
+
+type mockEC2Client struct {
+	ec2.Client
+	images       []ec2Types.Image
+	architecture string
+	err          error
+}
+
+func (m *mockEC2Client) DescribeImages(ctx context.Context, input *ec2.DescribeImagesInput, opts ...func(*ec2.Options)) (*ec2.DescribeImagesOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+	if m.architecture == "x86_64" {
+		return &ec2.DescribeImagesOutput{
+			Images: []ec2Types.Image{
+				{
+					Architecture: ec2Types.ArchitectureValues("x86_64"),
+				},
+			},
+		}, nil
+	} else if m.architecture == "" {
+		return &ec2.DescribeImagesOutput{
+			Images: m.images,
+		}, nil
+	} else {
+		return nil, errors.New("unexpected architecture")
+	}
+}
```

Comment:
Same note as the mock that's at the top of the page. 

You might also want to put the mocks together near the tests that uses them or together at the end of the file, but not half and half where there's some at the top and some at the bottom.